docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -55,6 +55,9 @@ init_kubernetes() {
                 echo "Trying to obtain LoadBalancer external endpoint..."
                 sleep 10
                 export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[0].ip" )
+                if [ -z "${PUBLISHED_ADDRESS}" ]; then
```

Comment:
Based on this comment from @Tristan1900 , is this change going to work on GKE? Does it work on PKS?

Reply:
Yes, it'll work both for: 
- PKS and GKE where the IP address is available
- AWS where the endpoint is accessible using the hostname

This is because if the IP address is available, that is set as the published address (line 57). So, for PKS and GKE the IP address shall be set as published address. 

If the IP address is not available, the new `if` condition will set it to the hostname in lines 58-60 (which will work in AWS): 
https://github.com/pravega/pravega/blob/445f8c583380aeaa10c5e5e36457bcaaaccad93e/docker/pravega/scripts/init_kubernetes.sh#L58-L60

As for your second question, I've tested it on Azure Kubernetes Service (AKS) where the IP address is available, like for PKS and GKE. I built a new image and set up a new AKS cluster and deployed Pravega with external access enabled. I verified that a client application that writes events to the cluster and then reads them back works successfully. Subba had already tested the code with AWS.

Reply:
yes, I see it working in AWS.